Sources_v3/Controllers/ChannelController.swift

```diff
@@ -176,6 +203,20 @@ public class ChannelControllerGeneric<ExtraData: ExtraDataTypes>: Controller, De
         setupEventObservers()
     }
 
+    private func startDBObserving(_ completion: ((_ error: Error?) -> Void)? = nil) {
+        _channelObserver.reset()
+        _messagesObserver.reset()
+
+        do {
+            try channelObserver.startObserving()
+            try messagesObserver.startObserving()
+        } catch {
+            log.error("Failed to perform fetch request with error: \(error). This is an internal error.")
+            callback { completion?(ClientError.FetchFailed()) }
+            return
+        }
+    }
```

Comment:
Why is this not just?
```swift
private func startDBObserving() throws {
    _channelObserver.reset()
    _messagesObserver.reset()
    
    try channelObserver.startObserving()
    try messagesObserver.startObserving()
}
```

Reply:
It's possible option but in that case I'll have to write `do-try-catch` in both `startUpdating` and `didSet`.

Reply:
Well, that's good, because changing the query is still part of the `startUpdating` flow. If the fetching fails after the cid is set, we should still report it to the caller of `startUpdating`. In the current implementation, we just log the error, but the completion never gets called.

Reply:
Covered in https://github.com/GetStream/stream-chat-swift/pull/399